docs/lesson-plans/week-4.md

```diff
@@ -0,0 +1,623 @@
+
+The api directory is where all requests will enter through. It is the way that any external program will interact with
+and view the results of our application.
+
+The persist directory is where all *persisting* information will live. Starting out, it is where we'll create our 
+database mocks, and later on, it will be where our database interactivity classes will be placed.
+
+The service directory is what unifies everything together. Since we want to not make the persist and api directories 
+dependent on each other, there will end up being a few interfaces that are made available and implemented in service. 
+This is because some of the operations we want to perform need to know of both persist and api functionality (but 
+remember, they shouldn't know about each other). 
+
+Don't worry about having to figure all of this out yourself though, we'll provide some help to get you started. Also,
+for those of you who know about the 
+[Model-View-Controller](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93controller) design pattern, this 
+follows that! In this case, persist is our model (since it stores and holds the data), api is our view (since it's what
+the user/external applications use), and service is our controller (since it links everything together and handles
+the interactions between them).
+
+>For those of you who don't know, Model View Controller (or MVC) is a design pattern which separates out functionality 
+>into three different main functionalities. 
+>
+>- Model: deals with handling the data and functionality with processing data
+>
+>- View: deals with presenting data (whether visually, textually, or in any other way) in a way that an external entity 
+>(like you or another program) can understand what is happening and interact with your program
+>
+>- Controller: deals with linking up interactions between the model and view, and is the mastermind behind running your 
+>program and handling external entity interactions (like you clicking a button or something else sending requests)
+
+### What is Routing?
+
+Routing is when we define the actual routes/paths of our application and the functionality that they provide. 
+
+![simple-url-example](./img/simple_url.png)
+
+*https://doepud.co.uk/blog/anatomy-of-a-url*
+
+In the image above, when we refer to the path, we're talking about the part from (and including) the slash after the 
+domain name. So, for example, in the [https://c4cneu.com/jumpstart](https://c4cneu.com/jumpstart) URL, `/jumpstart` is 
+the path. Another image with a more complex URL will be shown below which dissects more components.
+
+### Routing in Vert.x
+
+Routing in Vertx is done with `Router` and `Route` objects provided by the Vert.x dependency. In the api directory, we 
+have an `ApiMain` class, which handles creating a `Router` Vert.x object, setting up a lot of the routing settings we'll 
+need in the future, creating a home `Route`, and setting the port to start on.
+
```

Comment:
Maybe just add the ApiMain.java code here, or at least snippits so there's no ambiguity about what you're referring to